### DISCUSSION TITLE
Cloudflare Functions > [pages:err] ReferenceError: DATABASE_URL is not defined.

### DISCUSSION THREAD

**Discussion by ciocan** [prisma/prisma#15207] [Q&A]
Issue when running on Cloudflare Functions:

> [pages:err] ReferenceError: DATABASE_URL is not defined.

```Typescript
import { PrismaClient } from "@prisma/client/edge";

export async function onRequestGet({ env }) {
  console.log(env.DATABASE_URL);
  const prisma = new PrismaClient({
    datasources: {
      db: {
        url: env.DATABASE_URL,
      },
    },
  });

  return new Response(`ok`);
}
```

Running with:
    "prisma": "^4.3.1",
    "@prisma/client": "^4.3.1",

Running script:
    "dev": "wrangler pages dev ./dist",

The DATABASE_URL is added to wrangler secrets and set in .env / .dev.vars

Does anyone make it work with Cloudflare Functions?

**Comment by austincrim** [comment]
Hey @ciocan 👋, sorry for the delay in getting to your question. Let me know if you have solved this since posting.

What version of the `wrangler` CLI tool are you using?

**Reply by vladi-stevanovic** [reply]
👋 Hello @ciocan did you perhaps have a chance to check Austin's questions? 
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

**Reply by derkoe** [reply]
I'm having the same issue - DATABASE_URL is added to wrangler secrets and set in .env / .dev.vars. 

I'm running `wrangler 2.12.0`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
